libd/src/d/semantic/statement.d

```diff
@@ -387,29 +387,35 @@ public:
 		stop  = buildImplicitCast(pass, stop.location, type, stop);
 
 		BinaryOp cmp_op;
-		UnaryOp inc_op;
 
 		if (f.reverse) {
 			swap(start, stop);
-			cmp_op = BinaryOp.GreaterEqual;
-			inc_op = UnaryOp.PreDec;
+			cmp_op = BinaryOp.Greater;
 		} else {
 			cmp_op = BinaryOp.Less;
-			inc_op = UnaryOp.PreInc;
 		}
-		
+
 		auto idx = new Variable(iDecl.location, type.getParamType(iDecl.type.isRef, false), iDecl.name, start);
 
 		idx.step = Step.Processed;
 		currentScope.addSymbol(idx);
 
-		auto idxExpr = new VariableExpression(idx.location, idx);		
-		auto initialize = f.reverse 
-			? new ExpressionStatement(new UnaryExpression(loc, idx.type, UnaryOp.PreDec, idxExpr))
-			: new SymbolStatement(idx);
+		Expression idxExpr = new VariableExpression(idx.location, idx);
+		if (f.reverse) {
+			idxExpr = new UnaryExpression(loc, type, UnaryOp.PostDec, idxExpr);
+		}
+
+		auto initialize = new SymbolStatement(idx);
 		auto condition = new BinaryExpression(loc, Type.get(BuiltinType.Bool), cmp_op, idxExpr, stop);
-		auto increment = new UnaryExpression(loc, type, inc_op, idxExpr);
-		
+
+		Expression increment;
+		if (f.reverse) {
```

Comment:
and initialize both increment and condition in there.

You'll have one branch and no idden dependancy accross branches.